lib/opte/src/engine/packet.rs

```diff
@@ -2867,14 +2878,15 @@ mod test {
             ..Default::default()
         };
 
+        let ip4_total_len = Ipv4Hdr::BASE_SIZE + tcp.hdr_len() + body.len();
         let ip4 = Ipv4Meta {
             src: SRC_IP4,
             dst: DST_IP4,
             proto: Protocol::TCP,
             ttl: 64,
             ident: 99,
-            hdr_len: 20,
-            total_len: 40,
+            hdr_len: Ipv4Hdr::BASE_SIZE.try_into().unwrap(),
+            total_len: ip4_total_len.try_into().unwrap(),
```

Comment:
Were we not using this in a substantive way before? Seems crazy that a hardcoded `40` would work at all when the intended semantic of the `Ipv4Meta::hdr_len` field is to be the same as the actual IPv4 header.

Reply:
This function previously always returned an empty TCP packet, so an IPv4 total length of 40 (20 IPv4 Header + 20 TCP Header) was correct. I changed it to also accept a payload now hence removing the hardcoded values.